src/pepr/operator/controllers/config/config.ts

```diff
@@ -3,109 +3,276 @@
   }
 
   // Update other config values (no need for special handling)
-  UDSConfig.allowAllNSExemptions = decodedConfigData.UDS_ALLOW_ALL_NS_EXEMPTIONS === "true";
+  UDSConfig.allowAllNSExemptions = policy.allowAllNsExemptions === true;
+
+  configLog.info(
+    `${firstLoad ? "Loaded" : "Updated"} UDS Config based on uds-operator-config ClusterConfig${firstLoad ? "" : " change"}`,
+  );
+}
+
+// Loads the UDS Config on startup
+export async function loadUDSConfig() {
+  // Run in Admission and Watcher pods
+  if (process.env.PEPR_WATCH_MODE || process.env.PEPR_MODE === "dev") {
+    let cfg: ClusterConfig = {};
+    let cfgSecret: kind.Secret = {};
 
-  log.info("Updated UDS Config based on uds-operator-config secret changes");
+    try {
+      cfg = await K8s(ClusterConfig).Get("uds-cluster-config");
+      cfgSecret = await K8s(kind.Secret).InNamespace("pepr-system").Get("uds-operator-config");
+    } catch (e) {
+      configLog.warn(e, "Error while fetching cluster config and operator config secret");
+    }
+
+    if (!cfg) {
+      throw new Error("No ClusterConfig found");
+    }
+
+    try {
+      validateCfg(cfg);
+      await updateCfg(cfg);
+      await updateCfgSecrets(cfgSecret || {});
+      configLog.info(redactConfig(), "Loaded UDS Config");
+    } catch (e) {
+      configLog.error(e);
+      throw e;
+    }
+  }
+}
+
+// Helper function for redacting sensitive values from the UDS Config
+function redactConfig() {
+  const authserviceRedisUri = UDSConfig.authserviceRedisUri ? "****" : "";
+  return { ...UDSConfig, authserviceRedisUri };
+}
+
+// Helper function to detect if 2 lists of CIDRs are equal, irrespective of order
+function areKubeNodeCidrsEqual(newCidrs: string[] = [], currentCidrs: string[] = []): boolean {
+  if (newCidrs.length !== currentCidrs.length) {
+    return false;
+  }
+  const sortedNewCidrs = [...newCidrs].sort();
+  const sortedCurrentCidrs = [...currentCidrs].sort();
+  return sortedNewCidrs.every((cidr, index) => cidr === sortedCurrentCidrs[index]);
+}
+
+// Starts a watch of the cluster config, used for Admission pods
+export async function startConfigWatch() {
+  // only run in admission controller or dev mode
+  if (process.env.PEPR_WATCH_MODE === "false" || process.env.PEPR_MODE === "dev") {
+    const watchCfg: WatchCfg = {
+      resyncFailureMax: process.env.PEPR_RESYNC_FAILURE_MAX
+        ? parseInt(process.env.PEPR_RESYNC_FAILURE_MAX, 10)
+        : 5,
+      resyncDelaySec: process.env.PEPR_RESYNC_DELAY_SECONDS
+        ? parseInt(process.env.PEPR_RESYNC_DELAY_SECONDS, 10)
+        : 5,
+      lastSeenLimitSeconds: process.env.PEPR_LAST_SEEN_LIMIT_SECONDS
+        ? parseInt(process.env.PEPR_LAST_SEEN_LIMIT_SECONDS, 10)
+        : 300,
+      relistIntervalSec: process.env.PEPR_RELIST_INTERVAL_SECONDS
+        ? parseInt(process.env.PEPR_RELIST_INTERVAL_SECONDS, 10)
+        : 1800,
+    };
+    const watcher = K8s(ClusterConfig).Watch(async (cfg, phase) => {
```

Comment:
Add types to callback:

```suggestion
    const watcher = K8s(ClusterConfig).Watch(async (cfg: ClusterConfig, phase: WatchPhase) => {
```